File: Practice/27/c++/27/27/27.cpp
#include<iostream>
#include<vector>
#include<ctime>

void Coutline(std::vector<int>a, int max, int remember = 0) {
	for (remember; remember < max;remember++) {
		std::cout << a[remember] << " ";
	}
}

std::vector<int> BozoSort(std::vector<int>a,int max) {
	bool sorted = false;
	while (!sorted) {
		int x1 = rand() % max;
		int x2 = rand() % max;

		int swap = a[x1];
		a[x1] = a[x2];
		a[x2] = swap;

		sorted = true;
		
		for (int i = 1; i < max;i++) {
			if (a[i - 1] < a[i]) {
				sorted = false;
				break;
			}
		}
		
	}

	return a;
}


int main() {
	std::vector<int>line;
	std::vector<int>temp;
	int x, n;
	std::cin >> n;

	int max = 0;
	int remember = 0 ;
	for (int i = 0;i <n;i++) {
		std::cin >> x;
		line.push_back(x);
		max++;
		if (max <= 5) {
			temp = BozoSort(line, max);
			Coutline(temp,max, remember);
		}
		else {
			remember = max - 5;
			temp = BozoSort(line, max);
			Coutline(temp, max, remember);
		}
		std::cout <<std::endl ;
	}
}

File: Practice/27/python/27.py
import random


def Coutline(a, maxx, remember):
	for remember in range (remember, maxx):
		print(a[remember], end=" ")
	print()


def BozoSort(a, maxx):
	sorted = False
	while(sorted == False):
		x1 = random.randint(0,maxx-1)
		x2 = random.randint(0,maxx-1)

		swap = a[x1]
		a[x1] = a[x2]
		a[x2] = swap

		sorted = True
		for i in range (1,maxx):
			if (a[i-1]<a[i]):
				sorted = False
				break
	return a


n = int(input())
a = list(map(int, input().split()))
line =[]
temp=[]
maxx = 0
remember = 0
t = 0
for i in range (n):
	line.append(a[t])
	t+=1
	maxx+=1
	if (maxx<=5):
		temp = BozoSort(line, maxx)
		Coutline(temp,maxx, remember)
	else:
		remember = maxx - 5
		temp = BozoSort(line, maxx)
		Coutline(temp, maxx, remember)
